### ISSUE TITLE
Without `serial: 1` napalm_install_config fails due to disappearing candidate.cfg

### ISSUE THREAD

**Issue by cs-1** [napalm-automation/napalm-ansible#137]
Hi!

I use napalm from the pip repository, the current napalm-ansible development branch from github together with the napalm-aos community driver (development version) with Python 3.

When I run a playbook _without_ `serial: 1` I almost randomly (i. e. not for every task) encounter errors like
```
failed: [myswitchname] (item=/xxx/ansible-networking/configs/myswitchname/020-mst-merge.txt) => {"changed": false, "item": "/xxx/ansible-networking/configs/myswitchname/020-mst-merge.txt", "msg": "cannot load config: [Errno 2] No such file or directory: '/var/folders/1b/zt9ydwjj26l57xlrrn0hgjj80000gn/T/candidate.cfg'"}
```
as soon as I run several "napalm_install_config" tasks. This particular playbook currently runs for two hosts. If I add `serial: 1` to the playbook, everything works as expected. I think that the problem is that "candidate.cfg" is not a unique filename for each host and I believe that the "napalm_install_config" tasks that run in parallel probably delete "candidate.cfg" as soon as they finished while another task might expect to be working on its own "candidate.cfg". I was trying to figure out which part of napalm-ansible actually creates the "candidate.cfg" file. I suspect that it is the driver but I'm not entirely sure -- could you help me out? In any case, an easy fix would be to add some randomness to the name for each creation of the file to make it unique. I hope to be able to help with a fix as soon as I understand where the filename is set.

**Comment by cs-1** [comment]
I just realized that the name is the default name in napalm-aos if no candidate configuration filename is given. So the fix would be to either have napalm-ansible give a sufficiently unique name to `driver. load_merge_candidate(...)` or to make the default name that the napalm-aos driver uses more unique. What do you think would be a better way?

**Comment by ogenstad** [comment]
It's an issue in the AOS library, ask them to look at this code:

https://github.com/napalm-automation-community/napalm-aos/blob/master/napalm_aos/aos.py#L234-L242

**Comment by cs-1** [comment]
Yes. The code would actually work if they wouldn't give a filename. However, they set the filename to a default of "candidate.cfg" if there's none given when the driver is called and so the code snippet you mentioned never creates a unique name but uses "candidate.cfg" instead. I've opened an issue in the napalm-aos repo. Thanks for your quick reply and for confirming my findings.
